source/globalCommands.py

```diff
@@ -3284,6 +3286,40 @@ def script_braille_cycleShowMessages(self, gesture: inputCore.InputGesture) -> N
 		msg = _("Braille show messages %s") % ShowMessages(newValue).displayString
 		ui.message(msg)
 
+	@script(
+		# Translators: Input help mode message for cycle through braille show selection command.
+		description=_("Cycle through the braille show selection states"),
+		category=SCRCAT_BRAILLE
+	)
+	def script_braille_cycleShowSelection(self, gesture: inputCore.InputGesture) -> None:
+		"""Set next state of braille show selection and reports it with ui.message."""
+		featureFlag: FeatureFlag = config.conf["braille"]["showSelection"]
+		boolFlag: BoolFlag = featureFlag.enumClassType
+		values = [x.value for x in boolFlag]
+		currentValue = featureFlag.value.value
+		nextName: str = boolFlag(
+			currentValue % len(values) + 1
+		).name

```

Comment:
There's no need to split this on new lines, but if you are, might as well make it a named variable for clarity.

```suggestion
		nextValueIndex = (currentValue % len(values)) + 1
		nextName: str = boolFlag(nextValueIndex).name
```